src/Microsoft.Identity.Client/WamPublicClientApplication.cs

```diff
@@ -0,0 +1,167 @@
+
+#if SUPPORTS_WAM
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Security;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.ApiConfig.Executors;
+using Microsoft.Identity.Client.Core;
+
+namespace Microsoft.Identity.Client
+{
+    internal class WamPublicClientApplication : IPublicClientApplication
+    {
+        internal IServiceBundle ServiceBundle { get; }
+
+        internal WamPublicClientApplication(ApplicationConfiguration config)
+        {
+            ServiceBundle = Core.ServiceBundle.Create(config);
+        }
+
+        public bool IsSystemWebViewAvailable => false;
+
+        public IAppConfig AppConfig => ServiceBundle.Config;
+
+        public ITokenCache UserTokenCache => throw new PlatformNotSupportedException("No UserTokenCache on WAM");
+
+        public AcquireTokenInteractiveParameterBuilder AcquireTokenInteractive(IEnumerable<string> scopes)
+        {
+            return AcquireTokenInteractiveParameterBuilder.Create(
+                ClientExecutorFactory.CreateWamPublicClientExecutor(this),
+                scopes);
+        }
+
+        public AcquireTokenSilentParameterBuilder AcquireTokenSilent(IEnumerable<string> scopes, IAccount account)
+        {
+            return AcquireTokenSilentParameterBuilder.Create(
+                ClientExecutorFactory.CreateWamClientApplicationBaseExecutor(this),
+                scopes,
+                account);
+        }
+
+        public AcquireTokenSilentParameterBuilder AcquireTokenSilent(IEnumerable<string> scopes, string loginHint)
+        {
+            if (string.IsNullOrWhiteSpace(loginHint))
+            {
+                throw new ArgumentNullException(nameof(loginHint));
+            }
+
+            return AcquireTokenSilentParameterBuilder.Create(
+                ClientExecutorFactory.CreateWamClientApplicationBaseExecutor(this),
+                scopes,
+                loginHint);
+        }
+
+        // TODO(WAM): implement these
+        public Task<IAccount> GetAccountAsync(string identifier) => throw new NotImplementedException();
+        public Task<IEnumerable<IAccount>> GetAccountsAsync() => throw new NotImplementedException();
+        public Task RemoveAsync(IAccount account) => throw new NotImplementedException();
+
+        public AcquireTokenByIntegratedWindowsAuthParameterBuilder AcquireTokenByIntegratedWindowsAuth(IEnumerable<string> scopes) => throw new NotImplementedException();
+        public AcquireTokenByUsernamePasswordParameterBuilder AcquireTokenByUsernamePassword(IEnumerable<string> scopes, string username, SecureString password) => throw new NotImplementedException();
+
+        #region Unsupported With WAM
+        public AcquireTokenWithDeviceCodeParameterBuilder AcquireTokenWithDeviceCode(IEnumerable<string> scopes, Func<DeviceCodeResult, Task> deviceCodeResultCallback) => throw new NotImplementedException();
```

Comment:
>throw new NotImplementedException(); [](start = 171, length = 36)

Should we use the obsolete attribute instead to show a compiler error instead of a run-time error? I suppose we may have to throw a platformnotimplementedexception to follow the pattern used elsewhere allowing developers to work with interfaces instead of the concrete classes.

Reply:
this is not a public class ? so the obsolete attribute does not matter much here? (or do I mis-understand something?)